src/utils/framework.ts

```diff
@@ -182,10 +183,10 @@ export const createReportStructure = (
         }
 
         if (widget.widgetId === 'matrix2dWidget' && reportStructureVariant === DIMENSION_FIRST) {
-            if (!widget.properties) {
+            const { properties: { data } } = widget as Matrix2dWidgetElement;
```

Comment:
Don't we need to check whether or not properties is present?

Reply:
I believe the properties object will always be there.

Reply:
Let's make some minor change before we merge this
```suggestion
            const { properties: { data } = {} } = widget as Matrix2dWidgetElement;
```

Reply:
Let's add a check for properties